### PR TITLE
Parse Rich header PE

### PR DESCRIPTION
**Pull request by GustavoLCR**
#12889

### CODE REVIEW

**Comment by XVilka** [comment]
See also https://github.com/radare/radare2/issues/662 and https://github.com/radare/radare2/issues/921

File: libr/bin/p/bin_pe.c [hunk 1]
```diff
@@ -331,6 +331,12 @@ static void header(RBinFile *bf) {
 	rbin->cb_printf ("  SizeOfHeapCommit : 0x%x\n", bin->nt_headers->optional_header.SizeOfHeapCommit);
 	rbin->cb_printf ("  LoaderFlags : 0x%x\n", bin->nt_headers->optional_header.LoaderFlags);
 	rbin->cb_printf ("  NumberOfRvaAndSizes : 0x%x\n", bin->nt_headers->optional_header.NumberOfRvaAndSizes);
+	RListIter *it;
+	Pe_image_rich_entry *entry;
+	rbin->cb_printf ("RICH_FIELDS\n");
+	r_list_foreach (bin->rich_entries, it, entry) {
+		rbin->cb_printf ("\tProduct: %d Name: %s Version: %d Times: %d\n", entry->productId, entry->productName, entry->minVersion, entry->timesUsed);
```
Comment (Maijin) [inline libr/bin/p/bin_pe.c]: Don't forget JSON output

File: libr/bin/p/bin_pe64.c [hunk 2]
```diff
@@ -71,6 +71,12 @@ static void header(RBinFile *bf) {
 	rbin->cb_printf ("\tSizeOfHeapCommit : 0x%x\n", bin->nt_headers->optional_header.SizeOfHeapCommit);
 	rbin->cb_printf ("\tLoaderFlags : 0x%x\n", bin->nt_headers->optional_header.LoaderFlags);
 	rbin->cb_printf ("\tNumberOfRvaAndSizes : 0x%x\n", bin->nt_headers->optional_header.NumberOfRvaAndSizes);
+	RListIter *it;
+	Pe_image_rich_entry *entry;
+	rbin->cb_printf ("RICH_FIELDS\n");
+	r_list_foreach (bin->rich_entries, it, entry) {
+		rbin->cb_printf ("\tProduct: %d Name: %s Version: %d Times: %d\n", entry->productId, entry->productName, entry->minVersion, entry->timesUsed);
```
Comment (Maijin) [inline libr/bin/p/bin_pe64.c]: Don't forget JSON output

Comment (radare) [reply libr/bin/p/bin_pe64.c]: Dont use \t. Only spaces

File: libr/bin/format/pe/pe.c [hunk 3]
```diff
@@ -2068,6 +2068,289 @@ static Sdb* Pe_r_bin_store_resource_version_info(PE_VS_VERSIONINFO* vs_VersionIn
...
+		off -= sizeof (ut32);
+		data = r_buf_read_le32_at (bin->b, off) ^ mask;
+		entry->productId = data >> 16;
+		entry->minVersion = data & 0xFFFF;
+		entry->productName = _known_product_ids (entry->productId);
+		off -= sizeof (ut32);
+		r_list_append (bin->rich_entries, entry);
+	}
+}
+
```
Comment (radare) [inline libr/bin/format/pe/pe.c]: Empty lines

File: libr/bin/format/pe/pe.c [hunk 4]
```diff
@@ -2068,6 +2068,289 @@ static Sdb* Pe_r_bin_store_resource_version_info(PE_VS_VERSIONINFO* vs_VersionIn
...
+
+
 static char* _resource_lang_str(int id) {
 	switch(id) {
```
Comment (radare) [inline libr/bin/format/pe/pe.c]: Missing space before (

**Pull request merged**